dace/codegen/targets/cpu.py

```diff
@@ -189,14 +190,24 @@ def allocate_view(self, sdfg: SDFG, dfg: SDFGState, state_id: int,
         self._dispatcher.dispatch_allocate(sdfg, dfg, state_id, viewed_dnode,
                                            global_stream, allocation_stream)
 
+        # When emitting ArrayInterface, we need to know if this is a read or
+        # write variation
+        if viewed_dnode in dfg.source_nodes():
+            is_write = False
+        elif viewed_dnode in dfg.sink_nodes():
+            is_write = True
+        else:
+            is_write = None
```

Comment:
Could this occur? If this is set to `None` then it may raise an exception in `cpp.py`

Reply:
I'm not sure, not in any SDFG we've written until now at least. A slightly more robust solution would be to compare all sink/source nodes against `view_dnode.data` instead?